src/coreclr/vm/syncblk.cpp

```diff
@@ -1567,6 +1571,19 @@ AwareLock::EnterHelperResult ObjHeader::EnterObjMonitorHelperSpin(Thread* pCurTh
         return AwareLock::EnterHelperResult_Contention;
     }
 
+    // Initialize g_currentProcessCpuCount and g_currentProcessIsCpuQuotaLimited if they haven't been initialized yet
+    if (g_currentProcessCpuCount == 0)
+    {
+        GetCurrentProcessCpuCount();
+    }
+
+    _ASSERTE(g_currentProcessCpuCount > 0);
+    if (g_currentProcessIsCpuQuotaLimited)
+    {
+        g_SpinConstants.dwMonitorSpinCount = 0;
```

Comment:
Don't think this is necessary since spin-waiting is skipped anyway based on the condition above. However, there is another use of `g_SpinConstants.dwMonitorSpinCount` in `EnterEpilogHelper` that should also be preceded by a check for `g_currentProcessIsCpuQuotaLimited`.